js/xt.js

```diff
@@ -0,0 +1,1072 @@
+            };
+            let method = undefined;
+            if (market['linear']) {
+                method = 'usdtPublicGetFutureMarketV1PublicQAggTicker';
+            } else if (market['inverse']) {
+                method = 'coinPublicGetFutureMarketV1PublicQAggTicker';
+            }
+            const response = await this[method] (this.extend (request, params));
+            return this.parseTicker (response, market);
+        }
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        // TODO: Add swap tickers
+        const responseSpot = await this.spotPublicGetDataApiV1GetTickers (params);
+        // spot
+        //
+        // {
+        //     "ltc_usdt": {
+        //       "high": 106.99,
+        //       "moneyVol": 1589953.528784,
+        //       "rate": 4.3400,
+        //       "low": 97.51,
+        //       "price": 105.52,
+        //       "ask": 105.61,
+        //       "bid": 105.46,
+        //       "coinVol": 15507.7052
+        //     }
+        // }
+        //
+        const result = [];
+        const idsSpot = Object.keys (responseSpot);
+        for (let i = 0; i < idsSpot.length; i++) {
+            const id = idsSpot[i];
+            const market = this.market (id);
+            const ticker = this.safeValue (responseSpot, id);
+            result.push (this.parseTicker (ticker, market));
+        }
+        // SWAP commented out bcs spot and USDT pairs have the same ids
+        // const responseUSDT = await this.usdtPublicGetFutureMarketV1PublicQAggTickers (params);
+        // const responseCOIN = await this.coinPublicGetFutureMarketV1PublicQAggTickers (params);
+        // swap
+        //
+        // [
+        //    {
+        //    "t": 1664276374733,
+        //    "s": "bch_usdt",
+        //    "c": "119.14",
+        //    "h": "119.99",
+        //    "l": "113.82",
+        //    "a": "2173861",
+        //    "v": "2542448.2328",
+        //    "o": "115.87",
+        //    "r": "0.0282",
+        //    "i": "119.18",
+        //    "m": "119.16",
+        //    "bp": "119.14",
+        //    "ap": "119.2"
+        //    }
+        // ]
+        //
+        // const tickers = this.arrayConcat(responseUSDT['result'], responseCOIN['result']);
+        // for (let i = 0; i < tickers.length; i++) {
+        //     const ticker = tickers[i];
+        //     const market = this.market (this.safeString (ticker, 's'));
+        //     result.push (this.parseTicker (ticker, market));
+        // }
+        return result;
+    }
+
+    parseTicker (response, market) {
+        if (market['spot']) {
+            const percentage = this.safeString (response, 'rate');
+            const marketId = this.safeString (response, 'symbol');
+            market = this.safeMarket (marketId, market, '-');
+            const symbol = market['symbol'];
+            const timestamp = this.safeString (response, 'depthTime');
+            return this.safeTicker ({
```

Comment:
We have a family of `safeMethods` that allow lookups using different keys, so instead of duplicating the entire `ticker` structure, you can do this:

```Javascript
                const high = this.safeString2 (response, 'high', 'h'),
                const low =  this.safeString2 (response, 'low', 'l'),
```
If you need to search 3+ keys you can use the safeXN methods (with X = string, number, value, etc)

Example:
```Javascript
      const high = this.safeStringN(response, ['key1', 'key2', 'key3', 'key4', ...])
````